### DISCUSSION TITLE
Curl and JSON support during building

### DISCUSSION THREAD

**Discussion by Draghmar** [swoole/swoole-src#4620] [Q&A]
I've noticed that there are two parameters that can be added to the build process: `--enable-swoole-json` and `--enable-swoole-curl`. I can't find any info about what those actually bring, though. What are benefits of using them? Is this related to coroutine support? Is this speed related? Is there any reason to enable them? I do use both JSON functions as well as curl here and there.

**Comment by leocavalcante** [comment]
`--enable-swoole-json` adds the `swoole_substr_json_decode()` function which is a zero-copy (faster implementation of a) JSON decoder. More details at: https://wenda.swoole.com/detail/107587

`--enable-swoole-curl` enables the usage of native cURL functions asynchronously, they can be used inside Coroutines without blocking.
